app/upgrade_cudos.go

```diff
@@ -1818,26 +1818,25 @@ func moveGenesisDelegation(genesisData *GenesisData, fromDelegatorAddress, toDel
 		return fmt.Errorf("amount to move is greater than delegated amount")
 	}
 
-	destinationDelegations, destinationDelegatedAmount := getDelegationData(genesisData, toDelegatorAddress, validatorAddress)
-	if destinationDelegations == nil {
+	destinationValidatorsDelegations, destinationValidatorDelegatedAmount := getDelegationData(genesisData, toDelegatorAddress, validatorAddress)
+	if destinationValidatorsDelegations == nil {
 		// No destination delegations
-		newMap := NewOrderedMap[string, sdk.Int]()
-		newMap.Set(toDelegatorAddress, amount)
-		genesisData.Delegations.Set(toDelegatorAddress, newMap)
-	} else if destinationDelegatedAmount == nil {
-
+		destinationValidatorsDelegations = NewOrderedMap[string, sdk.Int]()
+		destinationValidatorsDelegations.Set(validatorAddress, amount)
```

Comment:
This is the actual fix - the most important change (`toDelegatorAddress` -> `validatorAddress`)